src/apps/workspaces/crud/user_applet_access.py

```diff
@@ -570,6 +570,35 @@ async def get_workspace_respondents(
 
         return data, total, ordering_fields
 
+    async def get_applet_respondents_total(self, applet_id: uuid.UUID) -> int:
+        query: Query = (
+            select(UserSchema.id)
+            .select_from(SubjectSchema)
+            .join(UserSchema, UserSchema.id == SubjectSchema.user_id, isouter=True)
+            .join(AppletSchema, AppletSchema.id == SubjectSchema.applet_id)
+            .join(
+                UserAppletAccessSchema,
+                and_(
+                    UserAppletAccessSchema.applet_id == SubjectSchema.applet_id,
+                    UserAppletAccessSchema.user_id == SubjectSchema.user_id,
+                    UserAppletAccessSchema.role == Role.RESPONDENT,
+                ),
+                isouter=True,
+            )
+            .where(
+                SubjectSchema.applet_id == applet_id,
+                SubjectSchema.soft_exists(),
+            )
+            .group_by(
+                UserSchema.id,
+                func.coalesce(UserSchema.id, func.gen_random_uuid()),
+            )
```

Comment:
Could you please help me understand what is being queried here? Specifically, I'm confused about the purpose of the group by clause.
Based on the method name, it seems like the query should return the total amount of subject records for the applet, including both full accounts and shell accounts, which would be count(subject.id).
If the query is supposed to count only respondents (full accounts only), the result should be count(user_applet_access.id).
I don't see any reason to group the results and then count by the aggregated result. Perhaps I'm misunderstanding something.

Reply:
yeah, the query was much more complex than needs to, I rewrote it much simpler